tools/llm_bench/task/image_generation.py

```diff
@@ -95,13 +95,85 @@ def run_image_generation(image_param, num, image_id, pipe, args, iter_data_list,
+        f'steps={nsteps}, width={image_width}, height={image_height}, guidance_scale={guidance_scale}'
+    )
+    result_md5_list = []
+    max_rss_mem_consumption = ''
+    max_uss_mem_consumption = ''
+    max_shared_mem_consumption = ''
+    if (args['mem_consumption'] == 1 and num == 0) or args['mem_consumption'] == 2:
+        mem_consumption.start_collect_memory_consumption()
+    additional_args = {}
+    if guidance_scale is not None:
+        additional_args["guidance_scale"] = guidance_scale
+    else:
+        if 'lcm-sdxl' in args['model_type']:
+            additional_args["guidance_scale"] = 1.0
+        if 'turbo' in args['model_name']:
+            additional_args["guidance_scale"] = 0.0
+    input_text_list = [input_text] * args['batch_size']
+    input_token_size = 0
+    if tokenizer:
+        input_data = tokenizer(input_text, return_tensors='pt')
+        input_data.pop('token_type_ids', None)
+        input_tokens = input_data['input_ids'] if 'input_ids' in input_data else input_data
+        input_token_size = input_tokens[0].numel()
+    if num == 0 and args["output_dir"] is not None:
+        for bs_idx, in_text in enumerate(input_text_list):
+            llm_bench_utils.output_file.output_image_input_text(in_text, args, image_id, bs_idx, proc_id)
+    start = time.perf_counter()
+    res = pipe.generate(input_text, num_inference_steps=nsteps, height=image_height, width=image_width, **additional_args).data
+    end = time.perf_counter()
+    if (args['mem_consumption'] == 1 and num == 0) or args['mem_consumption'] == 2:
+        mem_consumption.end_collect_momory_consumption()
+        max_rss_mem_consumption, max_shared_mem_consumption, max_uss_mem_consumption = mem_consumption.get_max_memory_consumption()
+        mem_consumption.clear_max_memory_consumption()
+    for bs_idx in range(args['batch_size']):
+        image = Image.fromarray(res[bs_idx])
+        rslt_img_fn = llm_bench_utils.output_file.output_gen_image(image, args, image_id, num, bs_idx, proc_id, '.png')
+        result_md5_list.append(hashlib.md5(Image.open(rslt_img_fn).tobytes(), usedforsecurity=False).hexdigest())
+    generation_time = end - start
+    iter_data = gen_output_data.gen_iterate_data(
+        iter_idx=num,
+        in_size=input_token_size * args['batch_size'],
+        infer_count=nsteps,
+        gen_time=generation_time,
+        res_md5=result_md5_list,
+        max_rss_mem=max_rss_mem_consumption,
+        max_shared_mem=max_shared_mem_consumption,
+        max_uss_mem=max_uss_mem_consumption,
+        prompt_idx=image_id,
+    )
+    iter_data_list.append(iter_data)
+    metrics_print.print_metrics(
+        num,
+        iter_data,
+        warm_up=(num == 0),
+        max_rss_mem=max_rss_mem_consumption,
+        max_shared_mem=max_shared_mem_consumption,
+        max_uss_mem=max_uss_mem_consumption,
+        stable_diffusion=None,
```

Comment:
why? Is it possibly to get detaild perf metrics from text2image pipeline?

Reply:
no, text2image does not currently provide perf metrics. You can only measure `generate` itself